rj_gameplay/stp/play/__init__.py

```diff
@@ -87,41 +90,45 @@ def assign_roles(
 
         used_robots = set()
         for tactic in self.prioritized_tactics:
-            # TODO: handle if tactic requests more roles than exist
-            #       by not assigning tactic at all!
-            #       and give some default behavior (later)
             robots_for_tactic = []
             for cost_fn, role in tactic.role_requests:
                 min_cost = 1e9
                 cheapest_robot = None
 
                 for robot in world_state.our_robots:
-                    if robot in used_robots or not robot.visible:
+                    if robot in used_robots or robot in robots_for_tactic:
+                        continue
+                    if not robot.visible:
                         continue
                     cost = cost_fn(robot, world_state)
                     if cost < min_cost:
                         min_cost = cost
                         cheapest_robot = robot
 
                 if cheapest_robot is None:
-                    # TODO: consider looping until success?
-                    raise RoleAssignFailure(
-                        f"RoleRequest ({role}, {cost_fn}) was not assigned"
-                    )
-
-                used_robots.add(cheapest_robot)
+                    print(f"RoleRequest ({role}, {cost_fn}) was not assigned")
```

Comment:
Any reason you got rid of the RoleAssignFailure?

Reply:
If I reintroduced it the program would crash but I assumed my code was supposed to allow play to continue and be able to handle too many role requests without throwing errors

Reply:
gotcha